cnf-certification-test/networking/suite.go

```diff
@@ -188,20 +191,38 @@ func testUndeclaredContainerPortsUsage(env *provider.TestEnvironment) {
 		failedPod := false
 		for listeningPort := range listeningPorts {
 			if put.ContainsIstioProxy() && netcommons.ReservedIstioPorts[int32(listeningPort.PortNumber)] {
-				tnf.ClaimFilePrintf("%s is listening on port %d protocol %s, but the pod also contains istio-proxy. Ignoring.", put, listeningPort.PortNumber, listeningPort.Protocol)
+				tnf.ClaimFilePrintf("%s is listening on port %d protocol %s, but the pod also contains istio-proxy. Ignoring.",
+					put, listeningPort.PortNumber, listeningPort.Protocol)
 				continue
 			}
 
 			if !declaredPorts[listeningPort] {
-				tnf.ClaimFilePrintf("%s is listening on port %d protocol %s, but that port was not declared in any container spec.", put, listeningPort.PortNumber, listeningPort.Protocol)
+				tnf.ClaimFilePrintf("%s is listening on port %d protocol %s, but that port was not declared in any container spec.",
+					put, listeningPort.PortNumber, listeningPort.Protocol)
 				failedPod = true
+				nonCompliantObjects = append(nonCompliantObjects,
+					testhelper.NewPodReportObject(put.Namespace, put.Name,
+						"Listening port was declared in no container spec", false).
+						SetType(testhelper.ListeningPortType).
+						AddField(testhelper.PortNumber, strconv.Itoa(listeningPort.PortNumber)).
+						AddField(testhelper.PortProtocol, listeningPort.Protocol))
 			}
+			compliantObjects = append(compliantObjects,
```

Comment:
Should this be inside an `else`?  Is it possible to get both a compliant and non-compliant object from the same `put`?

Reply:
fixed in https://github.com/test-network-function/cnf-certification-test/pull/1207